### ISSUE TITLE
Reduce type piracy by shifting methods to AbstractFFTs

### ISSUE THREAD

**Issue by jishnub** [JuliaApproximation/GenericFFT.jl#24]
https://github.com/JuliaApproximation/GenericFFT.jl/blob/2e55016b642f5bdc7324a064b81f16dca6b967e4/src/GenericFFT.jl#L24-L33

These should ideally be moved to AbstractFFTs to reduce the type-piracy in this package.

**Comment by daanhb** [comment]
Yes, that would be more appropriate. Although it may have side-effects, but even then it would be worth fleshing them out.
